compiler/src/dmd/funcsem.d

```diff
@@ -3400,3 +3400,115 @@ private void checkPrintfScanfSignature(FuncDeclaration funcdecl, TypeFunction f,
             p, funcdecl.toChars());
     }
 }
+
+/***************************************************
+ * Visit each overloaded function/template in turn, and call dg(s) on it.
+ * Exit when no more, or dg(s) returns nonzero.
+ *
+ * Params:
+ *  fstart = symbol to start from
+ *  dg = the delegate to be called on the overload
+ *  sc = context used to check if symbol is accessible (and therefore visible),
+ *       can be null
+ *
+ * Returns:
+ *      ==0     continue
+ *      !=0     done (and the return value from the last dg() call)
+ */
+extern (D) int overloadApply(Dsymbol fstart, scope int delegate(Dsymbol) dg, Scope* sc = null)
+{
+    Dsymbols visited;
+
+    int overloadApplyRecurse(Dsymbol fstart, scope int delegate(Dsymbol) dg, Scope* sc)
+    {
+        // Detect cyclic calls.
+        if (visited.contains(fstart))
+        return 0;
```

Comment:
Indentation is off throughout this function

Reply:
I think my rebase messed that up somehow.

Reply:
Fixed